apps/sms/services/js/drafts.js

```diff
@@ -46,11 +46,7 @@
       drafts.push(draft);
       draftIndex.set(draft.threadId, drafts);
 
-      this.store();
-
-      this.emit('saved', draft);
-
-      return this;
+      return this.store().then(() => this.emit('saved', draft));
```

Comment:
Right, I meant in this patch, only ```store``` will return promise so no need to worry about new event :) We'll need to decide when we'll be implementing ```DraftsService``` where both ```add``` and ```delete``` will be asynchronous.